packages/commercetools/theme/components/MyAccount/ShippingAddressForm.vue

```diff
@@ -0,0 +1,287 @@
+          v-slot="{ errors }"
+          class="form__element"
+        >
+          <SfInput
+            data-cy="shipping-details-input_zipCode"
+            v-model="form.zipCode"
+            name="zipCode"
+            label="Zip-code"
+            required
+            :valid="!errors[0]"
+            :errorMessage="errors[0]"
+          />
+        </ValidationProvider>
+        <ValidationProvider
+          :rules="`required|oneOf:${countries.map(c => c.name).join(',')}`"
+          v-slot="{ errors }"
+          class="form__element"
+        >
+          <SfSelect
+            data-cy="shipping-details-select_country"
+            class="form__select sf-select--underlined"
+            :value="form.country"
+            @selected="form.country = $event"
+            name="country"
+            label="Country"
+            required
+            :valid="!errors[0]"
+            :errorMessage="errors[0]"
+          >
+            <SfSelectOption
+              v-for="{ name, label } in countries"
+              :key="name"
+              :value="name"
+            >
+              {{ label }}
+            </SfSelectOption>
+          </SfSelect>
+        </ValidationProvider>
+      </div>
+      <ValidationProvider
+        rules="required|min:8"
+        v-slot="{ errors }"
+        class="form__element"
+      >
+        <SfInput
+          data-cy="shipping-details-input_phoneNumber"
+          v-model="form.phoneNumber"
+          name="phone"
+          label="Phone number"
+          required
+          :valid="!errors[0]"
+          :errorMessage="errors[0]"
+        />
+      </ValidationProvider>
+      <SfButton data-cy="shipping-details-btn_update" class="form__button">
+        {{ isNew ? "Add the address" : "Update the address" }}
+      </SfButton>
+    </form>
+  </ValidationObserver>
+</template>
+
+<script>
+import {
+  SfInput,
+  SfButton,
+  SfSelect,
+  SfCheckbox
+} from '@storefront-ui/vue';
+import { required, min, oneOf } from 'vee-validate/dist/rules';
+import { ValidationProvider, ValidationObserver, extend } from 'vee-validate';
+import { reactive } from '@vue/composition-api';
+import { getSettings } from '@vue-storefront/commercetools-api';
+import { userShippingGetters } from '@vue-storefront/commercetools';
+
+extend('required', {
+  ...required,
+  message: 'This field is required'
+});
+
+extend('min', {
+  ...min,
+  message: 'The field should have at least {length} characters'
+});
+
+extend('oneOf', {
+  ...oneOf,
+  message: 'Invalid country'
+});
+
+export default {
+  name: 'ShippingAddressForm',
+
+  components: {
+    SfInput,
+    SfButton,
+    SfSelect,
+    SfCheckbox,
+    ValidationProvider,
+    ValidationObserver
+  },
+
+  props: {
+    address: {
+      type: Object,
+      default: {}
+    },
+    isNew: {
+      type: Boolean,
+      required: true
+    }
+  },
+
+  setup(props, { emit }) {
+    const form = reactive({
+      id: userShippingGetters.getId(props.address),
```

Comment:
it's fine, but keep in mind that you are in the commercetools integration here, meaning you have custom-platform-specific form, it doesn't have to be agnostic

Reply:
It's needed because data from PRO package has different format than our mocks in OS version.